common/src/main/java/com/wynntils/features/chat/ChatMentionFeature.java

```diff
@@ -61,69 +61,44 @@ private Pattern buildPattern() {
     public void onChat(ChatMessageReceivedEvent e) {
         Component message = e.getMessage();
 
-        Matcher looseMatcher = mentionPattern.matcher(ComponentUtils.getUnformatted(message));
+        StyledText styledText = StyledText.fromComponent(message);
 
-        if (looseMatcher.find()) {
-            if (markMention.get()) {
-                e.setMessage(rewriteComponentTree(message));
-            }
-            if (dingMention.get()) {
-                McUtils.playSoundUI(SoundEvents.NOTE_BLOCK_PLING.value());
-            }
-        }
-    }
-
-    private Component rewriteComponentTree(Component comp) {
-        // Make a copy of the component without the siblings
-        MutableComponent curr = MutableComponent.create(comp.getContents()).withStyle(comp.getStyle());
-        // .getString() is used here as it gives formattingchars when those exist. It is needed for guild messages
-        // because wynn still uses legacy coloring for it.
-        String text = curr.getString();
-
-        // if current component has text check it for mentions and rewrite the component if necessary
-        if (!text.isEmpty()) {
-            curr = rewriteMentions(curr, text, comp.getStyle());
-        }
+        StyledTextPart partFinding = styledText.getPartFinding(mentionPattern, PartStyle.StyleType.NONE);
```

Comment:
Focusing on the usage of the new StyledText here, I'm not entirely happy. Githubs view of the PR is less than ideal, but I'll try to comment somewhere close to the code...

First of all, from what I can see you are not using partFinding here, but are really just checking if the entire message matches the mention pattern. Why not use the normal match() function for that?

Reply:
Or `getMatcher` as you called it. (We can discuss naming separately)

You are calling it in the "should highlight text" part anyway, I see now.

Reply:
The comments just below answer this. This is because we don't really want to go through the edge case of marking a ping, when it is in 2 separate components (there is no logical way that this can happen). So instead of checking the whole string, we try to find the first part.